pyscaffold/api/helpers.py

```diff
@@ -0,0 +1,372 @@
+# -*- coding: utf-8 -*-
+"""
+Useful functions for manipulating the action list and project structure.
+"""
+from __future__ import absolute_import
+
+import functools
+import sys
+from copy import deepcopy
+
+from six import string_types
+
+from ..exceptions import ActionNotFound
+from ..log import logger
+from ..structure import FileOp, define_structure
+
+logger = logger  # Sphinx workaround to force documenting imported members
+"""Logger wrapper, that provides methods like :obj:`~.ReportLogger.report`.
+See :class:`~.ReportLogger`.
+"""
+
+NO_OVERWRITE = FileOp.NO_OVERWRITE
+"""Do not overwrite an existing file during update
+(still created if not exists)
+"""
+
+NO_CREATE = FileOp.NO_CREATE
+"""Do not create the file during an update"""
+
+
+# -------- Project Structure --------
+
+def ensure(structure, path, content=None, update_rule=None):
+    """Ensure a file exists in the representation of the project tree
+    with the provided content.
+    All the parent directories are automatically created.
+
+    Args:
+        structure (dict): project representation as (possibly) nested
+            :obj:`dict`. See :obj:`~.merge`.
+        path (str or list): file path relative to the structure root.
+            The directory separator should be ``/`` (forward slash) if
+            present.
+            Alternatively, a list with the parts of the path can be
+            provided, ordered from the structure root to the file itself.
+            The following examples are equivalent::
+
+                'docs/api/index.html'
+                ['docs', 'api', 'index.html']
+        content (str): file text contents
+        update_rule: see :class:`~.FileOp`, ``None`` by default
+
+    Note:
+        Use an empty string as content to ensure a file is created empty.
+    """
+    # Ensure path is a list.
+    if isinstance(path, string_types):
+        path = path.split('/')
```

Comment:
Using '/' here is according to your documentation but could lead to user errors on windows. `os.sep` holds the right path separator. One could even think about using `https://docs.python.org/3/library/pathlib.html` (and it's Python2 backport) that allows a proper handling of paths.